Lib/_pyrepl/reader.py

```diff
@@ -316,6 +310,12 @@ def calc_screen(self) -> list[str]:
         pos -= offset
 
         prompt_from_cache = (offset and self.buffer[offset - 1] != "\n")
+
+        if self.can_colorize:
+            colors = list(gen_colors(self.get_unicode()))
```

Comment:
Nitpick:

```suggestion
            colors = tuple(gen_colors(self.get_unicode()))
```

Reply:
Needs to stay a list. `disp_str()` partially consumes the list since it's called line per line.